src/rapids_dependency_file_generator/cli.py

```diff
@@ -38,28 +45,23 @@ def main():
         help="path to YAML config file",
     )
 
-    inclusive_group = parser.add_argument_group("optional, but mutually inclusive")
-    inclusive_group.add_argument(
+    codependent_args = parser.add_argument_group("optional, but codependent")
+    codependent_args.add_argument(
         "--file_key",
         help="The file key from `dependencies.yaml` to generate",
     )
-    inclusive_group.add_argument(
+    codependent_args.add_argument(
         "--generate",
         help="The file type to generate",
         choices=[str(x) for x in [GeneratorTypes.CONDA, GeneratorTypes.REQUIREMENTS]],
     )
-    inclusive_group.add_argument(
-        "--cuda_version",
-        help="The CUDA version used for generating the output",
-    )
-    inclusive_group.add_argument(
-        "--arch",
-        help="The architecture used for generating the output",
+    codependent_args.add_argument(
+        "--matrix",
+        help='string representing which matrix combination should be generated. i.e. --matrix "cuda=11.5;arch=x86_64"',
```

Comment:
This should be `e.g.` and not `i.e.` but it might be better to write `such as`.
```suggestion
        help='string representing which matrix combination should be generated, such as `--matrix "cuda=11.5;arch=x86_64"`',
```